src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email">Email Invoice</ActionPopoverItem>
+      <ActionPopoverItem icon="print">Print Invoice</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await user.click(screen.getByText("Print Invoice"));
+
+  expect(screen.getByRole("button")).toBeFocused();
+});
+
+test("pressing enter on a menu item focuses the menu button", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email">Email Invoice</ActionPopoverItem>
+      <ActionPopoverItem icon="print">Print Invoice</ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await screen.getByRole("button", { name: "Print Invoice" }).focus();
+  await user.keyboard("{Enter}");
+
+  expect(screen.getByRole("button")).toBeFocused();
+});
+
+test("clicking a disabled menu item does not call its onClick handler", async () => {
+  const onClick = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email" onClick={() => onClick("email")}>
+        Email Invoice
+      </ActionPopoverItem>
+      <ActionPopoverItem icon="print" onClick={() => onClick("print")} disabled>
+        Print Invoice
+      </ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await user.click(screen.getByText("Print Invoice"));
+
+  expect(onClick).not.toHaveBeenCalled();
+});
+
+test("pressing enter on a disabled menu item does not call its onClick handler", async () => {
+  const onClick = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email" onClick={() => onClick("email")}>
+        Email Invoice
+      </ActionPopoverItem>
+      <ActionPopoverItem icon="print" onClick={() => onClick("print")} disabled>
+        Print Invoice
+      </ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await screen.getByRole("button", { name: "Print Invoice" }).focus();
+  await user.keyboard("{Enter}");
+
+  expect(onClick).not.toHaveBeenCalled();
+});
+
+test("clicking a disabled menu item does not close the menu", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <ActionPopover>
+      <ActionPopoverItem icon="email">Email Invoice</ActionPopoverItem>
+      <ActionPopoverItem icon="print" disabled>
+        Print Invoice
+      </ActionPopoverItem>
+    </ActionPopover>
+  );
+
+  await user.click(screen.getByRole("button"));
+  await user.click(screen.getByText("Print Invoice"));
+
+  expect(screen.getByRole("list")).toBeInTheDocument();
```

Comment:
suggestion: it might be better to assert that it's also visible here if possible rather than it's just in the DOM etc

```suggestion
  expect(screen.getByRole("list")).toBeVisible();
```